### ISSUE TITLE
Uuidm.of_string allows any garbage after a valid uuid

### ISSUE THREAD

**Issue by reynir** [dbuenzli/uuidm#11]
```OCaml
# Uuidm.of_string "de43a2a5-59df-4914-bf8f-6388489469f8-and-this-is-not-a-valid-uuid-anymore";;
- : Uuidm.t option = Some de43a2a5-59df-4914-bf8f-6388489469f8
```
While it may be useful in some cases to allow for anything after a valid uuid when parsing the behavior is undocumented and may surprise some.

**Comment by dbuenzli** [comment]
Indeed, that's not intended. Thanks for the report.

**Comment by dbuenzli** [comment]
Mmmh actually [there's](https://erratique.ch/software/uuidm/doc/Uuidm/index.html#fmt_ascsii) a `pos` argument. So it was likely designed so that you could extract it from a larger string. So maybe this should rather be a doc update.

**Comment by dbuenzli** [comment]
Fixed by #12.
